packages/flutter/lib/src/widgets/editable.dart

```diff
@@ -286,10 +286,18 @@ class RawInputLineState extends ScrollableState<RawInputLine> {
                                           ..type = config.keyboardType);
       _keyboardHandle.setEditingState(_keyboardClient.editingState);
       _keyboardHandle.show();
-    } else if (!focused && _isAttachedToKeyboard) {
-      _keyboardHandle.release();
-      _keyboardHandle = null;
-      _keyboardClient.clearComposing();
+    } else if (!focused) {
+      if (_isAttachedToKeyboard) {
+        _keyboardHandle.release();
+        _keyboardHandle = null;
+      }
+      if (_keyboardClient.inputValue.composing.isValid) {
+        _keyboardClient.clearComposing();
+        InputValue value = _keyboardClient.inputValue;
+        scheduleMicrotask(() {
+          config.onChanged(value);
```

Comment:
What happens if another onChanged call happens between when this mirotask is scheduled and when it runs?  We could overwrite newer information with older information.